modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -387,21 +387,37 @@ PatternedCartesianMeshGenerator::generate()
     {
       // throw an error message if the input mesh does not contain the required meta data
       if (!hasMeshProperty<Real>("pattern_pitch_meta", _input_names[i]))
-        mooseError("In PatternedCartesianMeshGenerator ",
-                   _name,
-                   ": the unit square input mesh does not contain appropriate meta data "
-                   "required for generating a core mesh. Involved input mesh: ",
-                   _input_names[i],
-                   "; metadata issue: 'pattern_pitch_meta' is missing.");
+        mooseError(
+            "In PatternedCartesianMeshGenerator ",
+            _name,
+            ": the unit square input mesh does not contain appropriate meta data "
+            "required for generating a core mesh. Involved input mesh: ",
+            _input_names[i],
+            "; metadata issue: 'pattern_pitch_meta' is missing. Note that "
+            "'generate_core_metadata' is set to true, which"
+            "means that the mesh generator is producing a core mesh by stitching the input "
+            "assembly meshes together. Therefore,"
+            "the input meshes must contain the metadata of assembly meshes, which can "
+            "usually be either automatically assigned "
+            "by using another PatternedCartesianMeshGenerator with 'generate_core_metadata' set as "
+            "false or manually assigned by AddMetaDataGenerator.");
```

Comment:
Maybe also add, "Alternatively, for assembly structures created with PCCMG, generate_core_data can be set to false in Patterned[Cart|Hex]MeshGenerator to avoid this error message."

Reply:
Actually, it is recommended to use `Hex|CartesianConcentricCircleAdaptiveBoundaryMeshGenerator` for assembly level mesh generation.